### DISCUSSION TITLE
Mass Goland wing

### DISCUSSION THREAD

**Discussion by alee269** [ImperialCollegeLondon/sharpy#235] [Q&A]
Hello, I am trying to create a dataset using SHARPy in order to create a surrogate model to minimize the mass and maximize the L/D with a flutter constraint. I'm having some problems to extract the mass value. Running SHARPy in the output I see:
"Linearising gravity terms...
	M = 0.54 kg
	X_CG A -> -0.81 0.37 0.00
Node  1 	-> B -0.040 -0.127 0.000
			-> A 0.439 0.394 -0.000
			-> G 0.438 0.394 -0.035
	Node mass:
		Matrix: 23.4620
Node  2 	-> B -0.040 -0.127 0.000
			-> A 0.798 0.895 -0.001
			-> G 0.796 0.895 -0.064
	Node mass:
		Matrix: 11.7310
Node  3 	-> B -0.040 -0.127 0.000
			-> A 1.157 1.395 -0.001
			-> G 1.154 1.395 -0.093
	Node mass:
		Matrix: 23.4620
....."
what does that 0.54 kg refer to? Thank you.

**Comment by sduess** [comment]
Hello,

M is the gravitational mass considered for the linearization of the gravitational forces. If you have a free-flying model, i.e. considering rigid-body motions, M is the total mass of your aircraft. Since you are looking at the Goland wing, I assume are working with a clamped model. Since you do not have rigid body motions for this case, the mass written to your log file is the mass of the clamped node. 

If you want to have a more detailed look at the code, the output is written in the linearise_gravity_forces function in [lingebm.py](https://github.com/ImperialCollegeLondon/sharpy/blob/main/sharpy/linear/src/lingebm.py).
 
I hope that helps. Let me know if you have any further questions. 

Best,
Stefanie

**Reply by alee269** [reply]
Hello Stefanie,

thank you very much for your help and consideration. I have another question: considering always the goland wing example, where can I find the cl and cd values? Many thanks.

**Reply by sduess** [reply]
No worries!

You can export the aerodynamic forces by calling the postprocessor "AeroForcesCalculator". Be sure to call the postprocessor, after computing the static aeroelastic solution, i.e. put the postprocessor after the "StaticCoupled" solver in SHARPy's flow list.

**Reply by alee269** [reply]
Done, thank you very much for your help. Also, I was wondering, how did you normalize the u_inf velocity? Compared to what? 

Many thanks.

**Reply by sduess** [reply]
Glad you figured it out.

You can find all parameters used for the normalization in the linear system assembly settings, more precisely in the 'ScalingDict' setting.

**Reply by alee269** [reply]
Thanks for your reply! 

For the ScalingDist Setting, just to double check that, I initialized u_inf at 60 m/s, rho at 1.02 and c_ref at 3.6576.
'ScalingDict': {'length': 0.5 * ws.c_ref,
'speed': u_inf,
'density': rho},
so my length used is half chord (1.8288), speed is 60m/s and rho is 1.02, correct?

Using u_inf at 30 m/s I had no problem but increasing it the kernel dies and I don't understand why.
Thank you in advance for your help.

**Reply by sduess** [reply]
I hope you figured it out. We would need more information about the exact error to help you.
